Source/WebCore/page/writing-tools/WritingToolsController.mm

```diff
@@ -827,7 +847,10 @@ static bool appliedCommandIsWritingToolsCommand(const Vector<Ref<WritingToolsCom
 
     document->selection().setSelection({ range });
 
-    document->editor().replaceSelectionWithText(replacementText, Editor::SelectReplacement::Yes, Editor::SmartReplace::No, EditAction::InsertReplacement);
+    {
+        WritingToolsEditingScope editingScope { *document };
+        document->editor().replaceSelectionWithText(replacementText, Editor::SelectReplacement::Yes, Editor::SmartReplace::No, EditAction::InsertReplacement);
```

Comment:
instead of doing this scope thing here, could/should it be done in the writing tools command itself?

Reply:
oh nvm I misread where this is

Reply:
You'll have to convince me why that's better. I prefer to keep everything in one place.

Undo/redo are not blocked since that has another meaning while Writing Tools is active.